sql/CMakeLists.txt

```diff
@@ -927,3 +928,36 @@ ADD_CUSTOM_TARGET(distclean
 ADD_CUSTOM_TARGET(show-dist-name
   COMMAND ${CMAKE_COMMAND} -E echo "${CPACK_PACKAGE_FILE_NAME}"
 )
+
+# If the repository was modified (via push or pull) or if the
+# mysql_githash.h.in changed rebuild the mysql_githash.h file
+SET(GITHASH_HEADER_SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/mysql_githash.h.in)
+SET(GITHASH_HEADER ${CMAKE_CURRENT_BINARY_DIR}/mysql_githash.h)
+
+
+SET(GITHASH_DEPENDS ${GITHASH_HEADER_SOURCE})
+SET(GITHASH_PERL_OPTS --git_root=${CMAKE_SOURCE_DIR} --file=${GITHASH_HEADER})
+SET(GITHASH_PERL_OPTS ${GITHASH_PERL_OPTS}
+                      --rocksdb_githash=none}
+                      --rocksdb_gitdate=none)
+
+IF(DEFINED MYSQL_GITHASH)
+  # If the hashes and dates are passed in add them to the parameters for
+  # the perl script
+  SET(GITHASH_PERL_OPTS ${GITHASH_PERL_OPTS}
+                        --mysql_githash=${MYSQL_GITHASH}
+                        --mysql_gitdate=${MYSQL_GITDATE})
+ELSE()
+  # Otherwise we want to add the .git/logs/HEAD file to the list of
+  # dependencies
+  SET(GITHASH_DEPENDS ${GITHASH_DEPENDS} ${CMAKE_SOURCE_DIR}/.git/logs/HEAD)
+ENDIF()
+
+SET(GITHASH_PERL ${CMAKE_SOURCE_DIR}/cmake/githash.pl)
+ADD_CUSTOM_COMMAND(OUTPUT ${GITHASH_HEADER}
+  COMMAND ${CMAKE_COMMAND} -E copy ${GITHASH_HEADER_SOURCE} ${GITHASH_HEADER}
+  COMMAND ${CMAKE_COMMAND} -E echo ${GITHASH_PERL} ${GITHASH_PERL_OPTS}
+  COMMAND perl ${GITHASH_PERL} ${GITHASH_PERL_OPTS}
```

Comment:
This command fails with:

```
You must specify none or all of the git hashes and dates
Usage:
    githash.pl [options]
```

I think you need to change the script to handle only adding the two rocksdb arguments.